src/main/java/de/hysky/skyblocker/skyblock/dwarven/WishingCompassSolver.java

```diff
@@ -0,0 +1,410 @@
+            if (!verifyLocation(playerZone, targetLocation)) {
+                location = MiningLocationLabel.CrystalHollowsLocationsCategory.UNKNOWN;
+            }
+            //offset the jungle location to its doors
+            if (location == MiningLocationLabel.CrystalHollowsLocationsCategory.JUNGLE_TEMPLE) {
+                targetLocation = targetLocation.add(JUNGLE_TEMPLE_DOOR_OFFSET);
+            }
+
+            CLIENT.player.sendMessage(Constants.PREFIX.get()
+                            .append(Text.translatable("skyblocker.config.mining.crystalsWaypoints.wishingCompassSolver.foundMessage").formatted(Formatting.GREEN))
+                            .append(Text.literal(location.getName()).withColor(location.getColor()))
+                            .append(Text.literal(": " + (int) targetLocation.getX() + " " + (int) targetLocation.getY() + " " + (int) targetLocation.getZ())),
+                    false);
+
+            //add waypoint
+            CrystalsLocationsManager.addCustomWaypoint(location.getName(), BlockPos.ofFloored(targetLocation));
+        }
+
+        //reset ready for another go
+        reset();
+    }
+
+    /**
+     * using the stating locations and line direction solve for where the location must be
+     */
+    protected static Vec3d solve(Vec3d startPosOne, Vec3d startPosTwo, Vec3d directionOne, Vec3d directionTwo) {
+        //convert format to get lines for the intersection solving
+        Vector3D lineOneStart = new Vector3D(startPosOne.x, startPosOne.y, startPosOne.z);
+        Vector3D lineOneEnd = new Vector3D(directionOne.x, directionOne.y, directionOne.z).add(lineOneStart);
+        Vector3D lineTwoStart = new Vector3D(startPosTwo.x, startPosTwo.y, startPosTwo.z);
+        Vector3D lineTwoEnd = new Vector3D(directionTwo.x, directionTwo.y, directionTwo.z).add(lineTwoStart);
+        Line line = new Line(lineOneStart, lineOneEnd, 1);
+        Line lineTwo = new Line(lineTwoStart, lineTwoEnd, 1);
+        Vector3D intersection = line.intersection(lineTwo);
+
+        //return final target location
+        if (intersection == null || intersection.equals(new Vector3D(0, 0, 0))) {
+            return null;
+        }
+        return new Vec3d(intersection.getX(), intersection.getY(), intersection.getZ());
+
+
+    }
+
+    private static ActionResult onBlockInteract(PlayerEntity playerEntity, World world, Hand hand, BlockHitResult blockHitResult) {
+        if (CLIENT.player == null) {
+            return null;
+        }
+        ItemStack stack = CLIENT.player.getStackInHand(hand);
+        //make sure the user is in the crystal hollows and holding the wishing compass
+        if (!Utils.isInCrystalHollows() || !SkyblockerConfigManager.get().mining.crystalsWaypoints.wishingCompassSolver || !Objects.equals(stack.getSkyblockId(), "WISHING_COMPASS")) {
+            return ActionResult.PASS;
+        }
+        if (useCompass()) {
+            return ActionResult.FAIL;
```

Comment:
I don't think we should be canceling player input?

Reply:
This is how NEU does it but i can remove the feature if you do not want it. I assumed it was the same as stopping someone drop a item or not click a slot in a gui.

Reply:
Yeah, it's fine then.